Swiften/MIX/MIX.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2017 Tarun Gupta
+ * Licensed under the simplified BSD license.
+ * See Documentation/Licenses/BSD-simplified.txt for more information.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_set>
+
+#include <boost/signals2.hpp>
+
+#include <Swiften/Base/API.h>
+#include <Swiften/JID/JID.h>
+#include <Swiften/Elements/Message.h>
+#include <Swiften/Elements/MIXJoin.h>
+#include <Swiften/Elements/MIXLeave.h>
+#include <Swiften/Elements/MIXPayload.h>
+#include <Swiften/Elements/MIXRetract.h>
+#include <Swiften/Elements/ErrorPayload.h>
+
+namespace Swift {
+    class SWIFTEN_API MIX {
+        public:
+            using ref = std::shared_ptr<MIX>;
+
+        public:
+            virtual ~MIX();
+
+            /**
+            * Join a MIX channel and subscribe to nodes.
+            */
+            virtual void joinChannelAndSubscribe(const JID &channel, const std::unordered_set<std::string>& nodes) = 0;
```

Comment:
An object which conforms to the `MIX` interface is supposed to represent a single client <-> MIX channel relation. So the channel is going to be the same for all methods and probably passed in the constructor.